apps/settings/js/modules/bluetooth/bluetooth_connection_manager.js

```diff
@@ -0,0 +1,959 @@
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     */
+    _init: function btcm__init() {
+      // Observe 'defaultAdapter' property for reaching default adapter.
+      AdapterManager.observe('defaultAdapter',
+                             this._onDefaultAdapterChanged.bind(this));
+      this._onDefaultAdapterChanged(AdapterManager.defaultAdapter);
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * updating adapter immediately.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Object BluetoothAdapter} newAdapter
+     * @param {Object BluetoothAdapter} oldAdapter
+     */
+    _onDefaultAdapterChanged:
+    function btcm__onDefaultAdapterChanged(newAdapter, oldAdapter) {
+      Debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+      Debug('_onDefaultAdapterChanged(): oldAdapter = ' + oldAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+
+      if (oldAdapter) {
+        // unwatch event since the old adapter is no longer usefull
+        this._unwatchProfilesStatuschanged(oldAdapter);
+        this._unwatchDefaultAdapterOnattributechanged(oldAdapter);
+      }
+
+      if (newAdapter) {
+        // watch event since the new adapter is ready to access
+        this._watchProfilesStatuschanged(newAdapter);
+        this._watchDefaultAdapterOnattributechanged(newAdapter);
+        // restore connection
+        if (newAdapter.state === 'enabled') {
+          this._restoreConnection();
+        }
+      } else {
+        // reset properties only
+        this._resetConnectionInfo();
+      }
+    },
+
+    /**
+     * Init cache of connected device and save it in cache.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @return {Promise}
+     */
+    _initConnectedDevicesInfo: function btcm__initConnectedDevicesInfo() {
+      if (!this._defaultAdapter) {
+        return Promise.reject('default adapter is not existed!!');
+      }
+      // Init connection status and profile from platform.
+      // Then, save these connected device information in cache.
+      return this._getConnectedDevicesWithHFPAndA2DP().then(
+      (connectedDevicesByProfile) => {
+        return this._constructDeviceItemsMap(connectedDevicesByProfile);
+      }, (reason) => {
+        Debug('_initConnectedDevicesInfo(): rejected in ' + 
+              '_getConnectedDevicesWithHFPAndA2DP');
+        return Promise.reject(reason);
+      });
+    },
+
+    /**
+     * The method will update each device item in maintaining map.
+     *
+     * @access private
+     * @memberOf BluetoothConnectionManager
+     * @param {Oject} connectedDevices
+     * @return {Promise}
+     */
+    _constructDeviceItemsMap:
+    function btcm__constructDeviceItemsMap(connectedDevices) {
+      Debug('_constructDeviceItemsMap(): connectedDevices = ' + 
```

Comment:
This looks like a synchronous function, so it is not required to return a promise.